packages/block-library/src/navigation/block-navigation-list.js

```diff
@@ -8,6 +13,100 @@ import {
 	useSelect,
 	useDispatch,
 } from '@wordpress/data';
+import { focus } from '@wordpress/dom';
+import { useRef } from '@wordpress/element';
+import { UP, DOWN, LEFT, RIGHT } from '@wordpress/keycodes';
+
+/**
+ * Return focusables in a row element, excluding those from other branches
+ * nested within the row.
+ *
+ * @param {Element} rowElement The DOM element representing the row.
+ *
+ * @return {?Array} The array of focusables in the row.
+ */
+function getRowFocusables( rowElement ) {
+	const focusablesInRow = focus.focusable.find( rowElement );
+
+	if ( ! focusablesInRow || ! focusablesInRow.length ) {
+		return;
+	}
+
+	return focusablesInRow.filter( ( focusable ) => {
+		return focusable.closest( '[role="treeitem"]' ) === rowElement;
+	} );
+}
+
+function NavigableTreeGrid( { children } ) {
```

Comment:
A short comment here explaining why we need this component and what pressing left/right/up/down does would be helpful.